include/thread_pool/thread_pool.hpp

```diff
@@ -151,12 +146,13 @@ inline GenericThreadPool<Task, Queue>& GenericThreadPool<Task, Queue>::operator=
 {
     if (this != &rhs)
     {
-        m_idle_workers = std::move(rhs.m_idle_workers);
-        m_workers = std::move(rhs.m_workers);
-        m_rouser = std::move(rhs.m_rouser);
         m_failed_wakeup_retry_cap = rhs.m_failed_wakeup_retry_cap;
         m_next_worker = rhs.m_next_worker.load();
-        m_num_busy_waiters = rhs.m_num_busy_waiters.load();
+        m_rouser = rhs.m_rouser;
```

Comment:
Instead of copy assigning and then setting rhs to nullptr you have to just move it:
m_rouser = std::move(rhs.m_rouser);

This will automatically set the rhs.m_rouser to nullptr